xml/vk.xml

```diff
@@ -1558,7 +1558,7 @@ server.
             <member><type>VkBool32</type>                         <name>persistentContent</name><comment>VK_TRUE if this is a "smart" display that supports self-refresh/internal buffering.</comment></member>
         </type>
         <type category="struct" name="VkDisplayPlanePropertiesKHR" returnedonly="true">
-            <member><type>VkDisplayKHR</type>                     <name>currentDisplay</name><comment>Display the plane is currently associated with.  Will be VK_NULL_HANDLE if the plane is not in use.</comment></member>
+            <member optional="true"><type>VkDisplayKHR</type>                     <name>currentDisplay</name><comment>Display the plane is currently associated with.  Will be VK_NULL_HANDLE if the plane is not in use.</comment></member>
```

Comment:
It is `returnedonly` which already implies `optional` (actually more than that, the value is ignored).

Reply:
The value is ignored by the implementation, but probably not by an application. For context, I'm trying to use `optional="true"` attributes to improve static safety of nullable vulkan handles in a generated binding via types (see https://github.com/MaikKlein/ash/pull/96), so cases where the implementation may return a null handle are expressly of interest, since an objective is to force the caller to check the value in those cases rather than assuming it's nonnull.

This seems compatible with the definition of the attribute given in [the registry documentation](https://www.khronos.org/registry/vulkan/specs/1.1/registry.html), and I don't see where `returnedonly` is given to imply optionality. Having this information available with fine-grained detail rather than treating any returned handle at all as possibly null would be nice.

Reply:
That's laudable, though it is a stretch that it applies to post-conditions.
Though there are precedences in the XML (especially flags). There is no Diff in validity, so I guess this is harmless too...

Reply:
> though it is a stretch that the attribute currently applies to post-conditions

I think this provides a pretty strong motivating case that it should. Regardless, this too is not the first example of such a pattern. Pre-existing optional members in returned-only structs include:
- `VkQueueFamilyProperties::queueFlags`
- `VkSparseImageFormatProperties::aspectMask` and `::flags`
- `VkMemoryType::propertyFlags`
- `VkImageFormatProperties::sampleCounts`

and many others.

Reply:
So the optional attribute is indeed, as @krOoze suggests, only really meant for validation on the implementation side. 

I'm not sure how comfortable I am with this as at some point I would like to wrench out all of the autovalidity from the XML and put it directly into the spec instead; if there's a vestigial utility of the optional attribute I don't know what we'd do with it. I thus would prefer this to be a separate attribute, rather than muddying the water on what's already there.

It's also worth pointing out that the optional attribute came before the readonly attribute; the examples above are just things I missed when I added the "readonly" attribute, and don't serve any functional purpose.